libpod/container_internal_linux.go

```diff
@@ -698,11 +702,37 @@ func (c *Container) makeBindMounts() error {
 			// If it doesn't, don't copy them
 			resolvPath, exists := bindMounts["/etc/resolv.conf"]
 			if exists {
-
 				c.state.BindMounts["/etc/resolv.conf"] = resolvPath
 			}
+
+			// check if dependency container has an /etc/hosts file
 			hostsPath, exists := bindMounts["/etc/hosts"]
-			if exists {
+
+			// if not, and the new container has hosts to add, set the hosts basis path to /etc/hosts
+			if !exists && (len(c.config.HostAdd) > 0 || (len(c.state.NetworkStatus) > 0 && len(c.state.NetworkStatus[0].IPs) > 0)) {
+				hostsPath = defaultHostFile
+			}
+			// only go through all of this if we want to add a hosts file
+			if len(hostsPath) > 0 {
```

Comment:
I'll let others way in.  I'm muddling if we should remove the checks except " if !exists {" on line 712 or not?  I think if a container doesn't have a hosts file we'd want to create it regardless?  @mheon thoughts?